File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int major= nums[0];
        int counter = 1 ;
        for(int i = 01; i < nums.size();i++)
            {
                
                if(nums[i]==major)
                    counter++;
            else if (nums[i]!=major && counter == 0)
            {
                major = nums[i];
                counter = 1;
            }
            else 
                counter --;
            }
        return major ;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {

        int majorElement=nums[0];
        int majorElementCount=1;

        for (int i = 1; i < nums.length; i++) {
            int number = nums[i];
            if (number == majorElement)
            {
                majorElementCount++;
            }
            else if (--majorElementCount < 0)
            {
                majorElement=number;
                majorElementCount=1;
            }
        }
        return majorElement;
    }
}